workflow/artifacts/git/git.go

```diff
@@ -84,6 +85,14 @@ func (g *ArtifactDriver) Delete(s *wfv1.Artifact) error {
 
 func (g *ArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string) error {
 	a := inputArtifact.Git
+
+	// Azure DevOps requires capabilities multi_ack / multi_ack_detailed,
+	// which are not fully implemented and by default are included in
+	// transport.UnsupportedCapabilities.
+	transport.UnsupportedCapabilities = []capability.Capability{
+		capability.ThinPack,
+	}
```

Comment:
so as I understand, this currently resets a global for all hosts. we should limit this to Azure repos, as this is otherwise advertising incorrect capabilities to other hosts as well (which can have some unexpected consequences / side-effects).

Ideally we should only set it for the request and then set it back, but I'm not sure if there's a way to pass the `transport` as a variable (as relying on a global would not be thread-safe).

Also we should be a bit more explicit in the comment. And I think the approach from https://github.com/kluctl/kluctl/pull/637 is a bit more future proof -- remove those two capabilities instead of resetting:
```suggestion
	// Azure DevOps requires multi_ack* capabilities which go-git does not currently support
	// Workaround: removing those from UnsupportedCapabilities allows clones to work (see https://github.com/go-git/go-git/pull/613)
	// TODO: remove this once https://github.com/go-git/go-git/issues/64 is fully fixed
    var newCaps []capability.Capability
	for _, c := range transport.UnsupportedCapabilities {
		if c == capability.MultiACK || c == capability.MultiACKDetailed {
			continue
		}
		newCaps = append(newCaps, c)
	}
	transport.UnsupportedCapabilities = newCaps
```